### PR TITLE
Dice rolling and material collecting methods

### PR DESCRIPTION
**Pull request by Evan-Kreutzwiser**
Added dice rolling and material collecting methods.

The dice roll is done with 2 rolls from 1-6 to replicate the number rarities that 2 dice give to the board game

Material collection is a separate method from the dice rolling, and searches all of the settlement nodes for settlements that the current player owns. For every settlement the player owns, it checks each tile the node references to compare the harvest number to the dice roll, and makes sure the thief is not on the tile. If the number matches and there is no thief, the material is added to the player's card list.

### CODE REVIEW

File: KrampitzKreutzwiserSettlersOfCatan/src/krampitzkreutzwisersettlersofcatan/GamePanel.java [hunk 1]
```diff
@@ -180,19 +180,63 @@ public void mouseClick(MouseEvent event) {
...
+        // Roll the first dice
+        roll = (int)(Math.random()*6) + 1;
+        // Roll the second dice and add to the total
+        roll += (int)(Math.random()*6) + 1;
+        
+        System.out.println("Rolled a " + roll);
+        
+        // Act on the dice roll
+        if (roll == 7) { // Move the thief on a 7
+            // TODO: Allow player to move thief
```
Comment (Tacitor) [inline KrampitzKreutzwiserSettlersOfCatan/src/krampitzkreutzwisersettlersofcatan/GamePanel.java:199]: We can just make the thief move to a random tile actually

File: KrampitzKreutzwiserSettlersOfCatan/src/krampitzkreutzwisersettlersofcatan/GamePanel.java [hunk 2]
```diff
@@ -180,19 +180,63 @@ public void mouseClick(MouseEvent event) {
...
     private void diceRoll() {
-    
+        int roll; // Holds the dice roll
+        
+        // Roll the first dice
+        roll = (int)(Math.random()*6) + 1;
+        // Roll the second dice and add to the total
+        roll += (int)(Math.random()*6) + 1;
+        
+        System.out.println("Rolled a " + roll);
```
Comment (Tacitor) [inline KrampitzKreutzwiserSettlersOfCatan/src/krampitzkreutzwisersettlersofcatan/GamePanel.java:195]: Should probably display this on screen somewhere

File: KrampitzKreutzwiserSettlersOfCatan/src/krampitzkreutzwisersettlersofcatan/GamePanel.java (hunk 1 above)
Comment (Evan-Kreutzwiser) [reply KrampitzKreutzwiserSettlersOfCatan/src/krampitzkreutzwisersettlersofcatan/GamePanel.java:199]: good idea
but until then we have the TODO

File: KrampitzKreutzwiserSettlersOfCatan/src/krampitzkreutzwisersettlersofcatan/GamePanel.java (hunk 2 above)
Comment (Evan-Kreutzwiser) [reply KrampitzKreutzwiserSettlersOfCatan/src/krampitzkreutzwisersettlersofcatan/GamePanel.java:195]: yeah

File: KrampitzKreutzwiserSettlersOfCatan/src/krampitzkreutzwisersettlersofcatan/GamePanel.java [hunk 3]
```diff
@@ -180,19 +180,63 @@ public void mouseClick(MouseEvent event) {
...
+                // Search the tiles around the node
+                for (int j = 1; j <= 3; j++) {
+                    // If the tile has the same harvest number
+                    // And doesnt have the thief on it
+                    if (settlement.getTile(j).getHarvestRollNum() == harvestNumber
+                            && settlement.getTile(j).hasThief() == false) {
+                        // Give the player the tile's resource
+                        cards[player].add(settlement.getTile(j).getType());
+                        System.out.println("Gave player " + player + " a type " 
+                                + settlement.getTile(j).getType() + " resource");
```
Comment (Tacitor) [inline KrampitzKreutzwiserSettlersOfCatan/src/krampitzkreutzwisersettlersofcatan/GamePanel.java:235]: Also, need to account for large settlements, they give two cards.

**Review by Tacitor — APPROVED** [review]
Pretty good but needs some additions in the morning.

**Pull request merged**